src/v/cloud_roles/types.h

```diff
@@ -123,4 +134,12 @@ using api_response_parse_result = std::variant<
 using credentials_update_cb_t
   = ss::noncopyable_function<ss::future<>(credentials)>;
 
+// Azure expects every request to Blob Storage to contain an
+// 'x-ms-version' header that specifies the API version to use.
+// This version is hardcoded in Redpanda to ensure that an API
+// version that we've tested with is used in field.
+//
+// Update this version to use a different storage API version.
```

Comment:
> // Update this version to use a different storage API version.

I realize this comment was copied verbatim, but can you expand on this? i'm not sure i understand what "Update this version to use a different storage API version." means. is it a TODO item?

Reply:
azure api supports versioning, and we built against this version. added some context in the comment